### PR TITLE
Sorter

### PR DESCRIPTION
**Pull request by gpellizzari-scottlogic**
Added the AuthorPostSorter class with tests and changed the PostSorter interface to ask for the SortOrder (Asc or Desc)

### CODE REVIEW

File: src/main/java/com/scottlogic/AuthorPostSorter.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+public class AuthorPostSorter implements PostSorter {
+    @Override
+    public List<UserPost> sort(List<UserPost> inputList, SortOrder sortOrder) {
+        if (inputList == null) {
+            return null;
+        } else if (inputList.isEmpty()) {
+            return inputList;
+        }
+        Collections.sort(inputList, Comparator.comparing(UserPost::getAuthor));
```
Comment (jluxford-scottlogic) [inline src/main/java/com/scottlogic/AuthorPostSorter.java:15]: Might be worth being carful with this approach, by adjusting the `inputList` itself you no longer have a safe function as the inputs are changed when it is run. This is generally a bad idea because if you start finding issues and this is the cause they are incredibly hard to debug

File: src/test/java/com/scottlogic/AuthorPostSorterTest.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+import java.time.ZoneOffset;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class AuthorPostSorterTest {
+
+    @Test
+    void sort_withNull_returnsNull() {
```
Comment (jluxford-scottlogic) [inline src/test/java/com/scottlogic/AuthorPostSorterTest.java:16]: Nice naming style with the tests, in general instead of providing just `sort` you might want to do `authorPostSort` or `authorPostSorter` as when a full suite of tests is run you sometimes just get the failing test name and the file it fails in is slightly hidden. As you are doing lots of `sort` functions you might end up with lots of similar names so being able to easily differentiate them is useful.

File: src/test/java/com/scottlogic/AuthorPostSorterTest.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        List<UserPost> actual = null;
+        actual = new AuthorPostSorter().sort(actual, SortOrder.ASC);
```
Comment (jluxford-scottlogic) [inline src/test/java/com/scottlogic/AuthorPostSorterTest.java:18]: In general you want to avoid side effects and mutation inside tests. Instead of setting `actual` to be `null` and then also setting it to be the sorted version having a variable that is `initialList` or something similar makes the tests significantly clearer. The style of testing is different to normal coding as it is all about maximizing readability rather than efficiency as they will be run a finite number of times, this means a slight efficiency loss doesn't have much impact.

File: src/test/java/com/scottlogic/AuthorPostSorterTest.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+                OffsetDateTime.of(2020, 3, 12, 13, 22, 12, 0, ZoneOffset.UTC),
+                "An example of a post \nwith lines breaks.", 3);
+
+        List<UserPost> actual = Arrays.asList(userPost1, userPost2, userPost3);
+        List<UserPost> expected = Arrays.asList(userPost1, userPost3, userPost2);
+        //act
+        actual = new AuthorPostSorter().sort(actual, SortOrder.DESC);
+        //asses
+        Assertions.assertEquals(expected, actual);
+    }
```
Comment (jluxford-scottlogic) [inline src/test/java/com/scottlogic/AuthorPostSorterTest.java:85]: Nice set of cases there are a few edge cases and extra tests that you could write here to get full coverage, see if you can come up with as many as possible to practice writing unit tests

**Review by jluxford-scottlogic — APPROVED** [review]
